rust/catalyst-voting/src/crypto/group/utils.rs

```diff
@@ -0,0 +1,49 @@
+//! Utility functions and entities
+
+/// A convenient macro to autogenerate a `std::ops` traits implementation as `Add`, `Sub`
+/// etc.
+///
+/// It is required to provide a basic implementation of the `std::ops` for reference
+/// types, and then you can call `std_ops_gen` to provide a the remaining implementation
+/// E.g.
+/// ```
+/// struct YourType;
+/// impl Mul<&YourType> for &YourType {
+/// type Output = YourType;
+
+/// fn mul(self, other: &YourType) -> YourType {
+///    unimplemented!()
+/// }
+/// }
+///
+/// std_ops_gen!(Add, add, YourType, YourType, YourType);
+/// ```
+macro_rules! std_ops_gen {
```

Comment:
how necessary is this macro? I find it hard to read considering all its used for is GE arithmetic.

Reply:
right now it is not actually used.
It was originally used in such places the https://github.com/input-output-hk/catalyst-core/blob/f17b5103f8529b5efe05ecb1564d182bbae136ed/src/chain-libs/chain-crypto/src/ec/ristretto255.rs#L265.
But for now seems implementation only for references enough, so yea, will remove this unused code.
e.g.
```Rust
impl Mul<&Scalar> for &Scalar {
}
```